### ISSUE TITLE
support for making clip skip a parameter in generate function for stable diffusion pipeline

### ISSUE THREAD

**Issue by cm5kZGV2MjAyM3B1YmdpdGh1YmFjYw** [huggingface/diffusers#3870]
**Is your feature request related to a problem? Please describe.**
I am always very frustrurated when other pretrained model creator created a model with a suggested or specified clip skip value,and I dont know how to use or this api even exist.

**Describe alternatives you've considered**
No idea.

**Additional context**
I think its a very a very common tweakable parameter in the webui stablediffusion repo

**Comment by patrickvonplaten** [comment]
Duplicate - see discussion here: https://github.com/huggingface/diffusers/issues/1721 

I don't think it makes much sense to skip clip embeddings on which the unet has not been trained? Or is this feature to fine-tune a model?

**Comment by cm5kZGV2MjAyM3B1YmdpdGh1YmFjYw** [comment]
personally i think as a user they just want reproducibility being guarenteed.
that's why some creator will include the prompt to get the preview picture which is quite common in 3rd party repository as like civitai and some certain outside* blocked cn site.

**Comment by cm5kZGV2MjAyM3B1YmdpdGh1YmFjYw** [comment]
bump,unimplemented

**Comment by patrickvonplaten** [comment]
Asking again here, what is a common use case of this feature? Does one skip the final clip layer just for training/fine-tuning or also for inference?

**Comment by cm5kZGV2MjAyM3B1YmdpdGh1YmFjYw** [comment]
as far as i know its commonly used setting for users that uses kohya-ss.
They set that both during training and inference.
So right now,the model can be loaded but I am just not certain if it will perfectly reproduce the results expected by the original creator.

From my experience,there is some difference.with same prompt,model,lora,cfg,scheduler,seed
But of course our diffusers lora still havent stabilized so my observation could be wrong also.

**Comment by patrickvonplaten** [comment]
Also related: https://github.com/huggingface/diffusers/issues/3212

Think it's time to implement clip_skip!

**Comment by alexblattner** [comment]
>I don't think it makes much sense to skip clip embeddings on which the unet has not been trained? Or is this feature to fine-tune a model?

This attitude is why this repository is significantly behind A111. You had a similar attitude towards Lycoris too. Why are you skeptical to these features and not new models that aim to replace SD1.5? This makes no sense.

>Asking again here, what is a common use case of this feature? Does one skip the final clip layer just for training/fine-tuning or also for inference?

I would recommend looking at civitai.com and implement everything that is available as a start. They are far bigger than huggingface for stable diffusion model/lora/TI/etc sharing and for a good reason.

**Comment by alexblattner** [comment]
>I don't think it makes much sense to skip clip embeddings on which the unet has not been trained? Or is this feature to fine-tune a model?

This attitude is why this repository is significantly behind A111. You had a similar attitude towards Lycoris too. Why are you skeptical to these features and not new models that aim to replace SD1.5? This makes no sense.

>Asking again here, what is a common use case of this feature? Does one skip the final clip layer just for training/fine-tuning or also for inference?

I would recommend looking at civitai.com and implement everything that is available as a start. They are far bigger than huggingface for stable diffusion model/lora/TI/etc sharing and for a good reason.

@patrickvonplaten

**Comment by cm5kZGV2MjAyM3B1YmdpdGh1YmFjYw** [comment]
chill people,hahaha.
anyway since this issue is similar to https://github.com/huggingface/diffusers/issues/3212
we might as well close this thread and just make everything focuses on one thread and just make sure it will never be closed,lol.
